code/script.js

```diff
@@ -255,71 +273,180 @@ const checkQuestion = () => {
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+  if (category === 'Skin Color' || category === 'Skin/body texture') {
+    if (category === 'Skin Color') {
+      if (secret.skinColor === value) {
+        filterCharacters(true)
+      }
+      else {
+        filterCharacters(false)
+      }
+    }
+    else {
+      if (secret.skinTexture === value) {
+        filterCharacters(true)
+      }
+      else {
+        filterCharacters(false)
+      }
+    }
 
-  } else if (category === 'accessories' || category === 'other') {
+  } else if (category === 'Attributes' || category === 'Other attributes') {
+
+    if (category === 'Attributes') {
+      if (secret.attributes.includes(value)) {
+        filterCharacters(true)
+      }
+      else {
+        filterCharacters(false)
+      }
+    }
+    else {
+      if (secret.other.includes(value)) {
+        filterCharacters(true)
+      }
+      else {
+        filterCharacters(false)
+      }
+    }
 
   }
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
+  let stringPropertyName = ''  //Mappa (translate) Skin color och Skin texture
+  let arrayPropertyName = '' //Mappa (translate) Attribute and Other attributes
```

Comment:
Interesting approach. Different from mine.

Reply:
Yeah! I did this because it was easier for me to divide my arrays and strings in differens methods.